### PR TITLE
Correct flash memory layouts

### PR DESCRIPTION
**Pull request by NickeZ**
Document the actual values of the flash partition offsets and update linker scripts to match.

### CODE REVIEW

**Comment by NickeZ** [comment]
Please double check my math :)

File: bootloader.ld [hunk 1]
```diff
@@ -35,8 +35,7 @@ SEARCH_DIR(.)
 /* Memory Spaces Definitions */
 MEMORY
 {
-  /* Reserve (at least) 32 bytes of space for factory install random bytes */
-  rom      (rx)  : ORIGIN = 0x00000000, LENGTH = 0x0000DFE0
+  rom      (rx)  : ORIGIN = 0x00000000, LENGTH = 0x0000E000
```
Comment (benma) [inline bootloader.ld:38]: the previous value was correct - the last 32 bytes are reserved for the factory randomness:

https://github.com/digitalbitbox/bitbox02-firmware/blob/1bcb6dd934635a454d5ddb13524da569ec5ea1e3/src/random.c#L96

That should maybe be put into its own definition in flags.h.

Please update the diagram too (reduce bootloader area by 32 bytes, add the 32 bytes factory randomness item).

File: firmware.ld [hunk 2]
```diff
@@ -35,7 +35,7 @@ SEARCH_DIR(.)
 /* Memory Spaces Definitions */
 MEMORY
 {
-  rom      (rx)  : ORIGIN = 0x00010000, LENGTH = 0x000F0000
+  rom      (rx)  : ORIGIN = 0x00010000, LENGTH = 0x000D8000
```
Comment (benma) [inline firmware.ld:38]: i re-computed and came to the same conclusion.

**Review by benma — CHANGES_REQUESTED** [review]

File: bootloader.ld (hunk 1 above)
Comment (NickeZ) [reply bootloader.ld:38]: Interesting! I'll update it

**Review by benma — APPROVED** [review]
thanks, that was a very good and important catch!

**Pull request merged**